### PR TITLE
code style/quality rules

### PR DESCRIPTION
**Pull request by steinsiv**
## Further comments
Automatic remediation of warnings from code quality and language rules

See https://github.com/equinor/witsml-explorer/blob/main/.editorconfig for details

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -108,6 +108,10 @@ dotnet_remove_unnecessary_suppression_exclusions = none
 #### C# Coding Conventions ####
 [*.cs]
 
+# Turn the following off:
+dotnet_diagnostic.IDE0058.severity = none
+dotnet_diagnostic.xUnit1004.severity = none
```
Comment (janburak) [inline .editorconfig:115]: Is it possible to turn off xUnit1004 only for integration tests?

File: Src/Witsml/Data/Curves/DateTimeIndex.cs [hunk 2]
```diff
@@ -5,24 +5,22 @@ namespace Witsml.Data.Curves
...
+        public DateTime Value => _dateTime;
 
         public static DateTimeIndex FromString(string dateString)
         {
-            if (DateTime.TryParseExact(dateString, IsoPattern, null, DateTimeStyles.None, out var parsedDateTime))
-            {
-                return new DateTimeIndex(parsedDateTime);
-            }
-            throw new Exception($"Date format not recognized: {dateString}");
+            return (DateTime.TryParseExact(dateString, IsoPattern, null, DateTimeStyles.None, out var parsedDateTime))
```
Comment (janburak) [inline Src/Witsml/Data/Curves/DateTimeIndex.cs:21]: Unnecessary parentheses, do we ignore IDE0047?

Comment (steinsiv) [reply Src/Witsml/Data/Curves/DateTimeIndex.cs:21]: No I don't think so. 
It seems to me other warnings take precedence, when they are corrected IDE0047 will probably signal

**Review by janburak — APPROVED** [review]
- [] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Pull request merged**